nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -165,6 +165,9 @@ def _train(self):
                 # Check if we should stop the experiment
                 ret[DONE] = self._should_stop()
 
+                # Log result.
+                ray.get([w.log_result.remote(ret) for w in self.procs])
```

Comment:
This back-and-forth between processes seems unnecessary. It would be best to log from this process, rather than sending the result back to all the worker processes. One way to do this would be to put a classmethod on the `experiment_class`, and let the mixins override that classmethod.

Reply:
You make good points. I can remove the unnecessary back and forth between processes.

The main benefit I saw with this mixin was to simply reduce our dependence on ray. I'm not sure if multi-samples are supported in the way I implemented it. I can try and find out.

Reply:
One option: Maybe we should have a single class for interacting with wandb, and we could have two ways of using that class: via a mixin, or via a Ray logger. Both the mixin and the Ray logger would be thin wrappers around the class.

Reply:
I've thought about that, but wanted to keep this PR simple for now - to see if I can get a reasonable mixin working, and then refactor later.

Reply:
There's one other benefit in logging on the trainable btw, you get to initialize wandb on a worker process (in this case, I restricted it to rank 0). I wanted this scenario so that I could call `wandb` from other places in the code, and not have to funnel the logs to the result dict. I thought this would be very handy for, say, debugging a module (you could log grads via a backwards hook with no extra plumbing). 

I see the issue with passing all the logs back to the workers (in terms of reliability and unnecessary overhead). I think I'll go with your suggestion of using class methods, but include an option for the user to instantiate wandb on any of the worker nodes for extra logging - those that, by design, won't need to be passed back and forth in the result dict.